src/PackageSelection/PackageSelection.php

```diff
@@ -212,6 +223,10 @@ public function select(Composer $composer, bool $verbose): array
 
         $this->setSelectedAsAbandoned();
 
+        if(!empty($this->input->getOption('versions-only'))) {
```

Comment:
this should not use the input object here, as it would limit the usage of this class to commands defining such an option, which is out of its control. Pass the boolean to the constructor instead (or put it in the config)